src/renderer/features/wallet-connect-wallet-pairing/components/PairingModal.tsx

```diff
@@ -1,60 +1,92 @@
+import { Button, HeaderTitleText, Icon, type IconNames, SmallTitleText, StatusModal } from '@/shared/ui';
 import { Animation } from '@/shared/ui/Animation/Animation';
 import { Carousel, Modal } from '@/shared/ui-kit';
 import { EXPIRE_TIMEOUT, Step } from '../lib/constants';
-import { pairingForm } from '../model/pairingForm';
+import { type WalletTypeName } from '../lib/types';
+import { pairingFormModel } from '../model/form';
 
-import { ManageStep } from './ManageStep';
+import { PairingForm } from './PairingForm';
 import { WalletConnectQrCode } from './WalletConnectQrCode';
 import novawallet_onboarding_tutorial from './assets/novawallet_onboarding_tutorial.mp4';
 import novawallet_onboarding_tutorial_webm from './assets/novawallet_onboarding_tutorial.webm';
 
+const showCompleteStatus = (t: TFunction, showStatus: (props: StatusModalProps) => Promise<void>) => {
+  const WalletLogo: Record<WalletTypeName, IconNames> = {
+    [WalletType.WALLET_CONNECT]: 'walletConnectOnboarding',
+    [WalletType.NOVA_WALLET]: 'novaWalletOnboarding',
+  };
+
+  return (walletName: string, type: WalletTypeName) => {
+    showStatus({
+      title: walletName,
+      description: t('onboarding.walletConnect.pairedDescription'),
+      content: (
+        <div className="flex h-20 items-center justify-center gap-1">
+          <Icon name="logo" size={56} />
+          <div className="h-0 w-3 rounded border-[1.5px] border-text-positive"></div>
+          <Icon name="checkmarkOutline" className="text-text-positive" size={18} />
+          <div className="h-0 w-3 rounded border-[1.5px] border-text-positive"></div>
+          <Icon name={WalletLogo[type]} size={56} />
+        </div>
+      ),
+    });
+  };
+};
+
 type Props = PropsWithChildren<{
   variant: 'novawallet' | 'walletconnect';
 }>;
 
 export const PairingModal = memo(({ variant, children }: Props) => {
   const { t } = useI18n();
 
-  const session = useUnit(pairingForm.$session);
-  const uri = useUnit(pairingForm.$uri);
-  const step = useUnit(pairingForm.$step);
-  const open = useUnit(pairingForm.flow.state) === variant;
+  const { showStatus } = useStatusContext();
+
+  const session = useUnit(pairingFormModel.$session);
+  const uri = useUnit(pairingFormModel.$uri);
+  const step = useUnit(pairingFormModel.$step);
+  const state = useUnit(pairingFormModel.flow.state);
+
+  const open = state.type === variant;
 
   useEffect(() => {
     if (!open) return;
 
     const timeout = setTimeout(() => toggleModal(false), EXPIRE_TIMEOUT);
 
-    return () => clearTimeout(timeout);
+    return () => {
+      clearTimeout(timeout);
+    };
   }, [open]);
 
-  const goToScan = () => {
-    pairingForm.reset();
-    pairingForm.flow.open(variant);
-  };
-
   const toggleModal = (open: boolean) => {
     if (open) {
-      pairingForm.flow.open(variant);
+      pairingFormModel.flow.open({ type: variant, onComplete: showCompleteStatus(t, showStatus) });
```

Comment:
In order to use external function inside effector after we actually created new wallet